Cython/Compiler/TypeStubGenerator.py

```diff
@@ -0,0 +1,425 @@
+
+# Decided to revert to an older variant I had wrote of this code for the sake of 
+# maintainability - Vizonex
+
+
+
+# TODO Save this implemenation commented out if required....
+if sys.version_info >= (3, 9):
+    typing_module = "typing"
+else:
+    typing_module = "typing_extensions"
+
+
+
+class PyiWriter(CythonTransform, DeclarationWriter):
+    """Used By Cython to help Write stubfiles
+    this comes in handy for ides like Pylance 
+    which suffer from having no code acess to 
+    annotations from compiled python modules...
+    """
+
+    def __init__(self, context:Context):
+        super(PyiWriter, self).__init__(context=context)
+        super(DeclarationWriter, self).__init__()
+        self.context = context
+        self.module_name = ""
+        self.class_func_count = 0 
+
+        self.translation_table = {}
+        """Used as an eternal resource for translating ctype declarations into python-types"""
+
+        self.use_typing = False
+        """if true we must import typing's generator typehint..."""
+
+
+    def _visitchildren_indented(self, node):
+        self.indent()
+        self.visitchildren(node)
+        self.dedent()
+    
+    def translate_pyrex_type(self, ctype):
+        # TODO implement Pyrex to cython shadow typehints converter...
+        
+        if isinstance(ctype, PyrexTypes.CIntType):
+            return "int"
+
+        elif isinstance(ctype, PyrexTypes.CFloatType):
+            return "float"
+    
+        elif isinstance(ctype,PyrexTypes.PyObjectType):
+            return ctype.py_type_name()
+            
+        return 'object'
+
+
+    # Instead of doing it into C, we're doing it backwards...
+    def translate_base_type_to_py(
+        self,
+        base
+        ):
+
+        # Try checking our table first...
+        if self.translation_table.get(base.name):
+            return self.translation_table[base.name]
+
+        elif base.name == "object":
+            return "object"
+
+        elif base.name in ("unicode","basestring"):
+            return "str"
+
+        elif not base.is_basic_c_type:
+            # Likely that it's already a python object that's being handled...
+            # except for basestring and unicode...
+            return base.name 
+
+        elif base.name == "bint":
+            return "bool"
+
+        ctype = PyrexTypes.simple_c_type(base.signed, base.longness, base.name) # type: ignore
+        return self.translate_pyrex_type(ctype)
+
+    def emptyline(self):
+        self.result.putline("")
+
+    def visit_ModuleNode(self, node):
+        # We need to extract the name to write our pyi file down...
+        if node.directives['write_stub_file']:
+            result = self.write(node, True)
+            print("writing file %s.pyi ..." % node.full_module_name)
+            with open_new_file(os.path.join(node.full_module_name + '.pyi')) as w:
+                w.write("\n".join(result.lines))
```

Comment:
I'd always make sure that files end with a newline.
```suggestion
                w.write("\n".join(result.lines))
                w.write("\n")
```

Reply:
Smart Idea actually. Why did I not think of that before.